integrationTests/multiShard/relayedTx/relayedTx_test.go

```diff
@@ -0,0 +1,440 @@
+	for i := int64(0); i <= roundToPropagateMultiShard; i++ {
+		round, nonce = integrationTests.ProposeAndSyncOneBlock(t, nodes, idxProposers, round, nonce)
+		integrationTests.AddSelfNotarizedHeaderByMetachain(nodes)
+	}
+
+	time.Sleep(time.Second)
+	receiver1 := getUserAccount(nodes, receiverAddress1)
+	receiver2 := getUserAccount(nodes, receiverAddress2)
+
+	finalBalance := big.NewInt(0).Mul(big.NewInt(int64(len(players))), big.NewInt(nrRoundsToTest))
+	finalBalance.Mul(finalBalance, sendValue)
+	assert.Equal(t, receiver1.GetBalance().Cmp(finalBalance), 0)
+	assert.Equal(t, receiver2.GetBalance().Cmp(finalBalance), 0)
+
+	players = append(players, relayer)
+	checkPlayerBalances(t, nodes, players)
+}
+
+func TestRelayedTransactionInMultiShardEnvironmentWithSmartContractTX(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	nodes, idxProposers, players, relayer, advertiser := createGeneralSetupForRelayTxTest()
+	defer func() {
+		_ = advertiser.Close()
+		for _, n := range nodes {
+			_ = n.Messenger.Close()
+		}
+	}()
+
+	sendValue := big.NewInt(5)
+	round := uint64(0)
+	nonce := uint64(0)
+	round = integrationTests.IncrementAndPrintRound(round)
+	nonce++
+
+	receiverAddress1 := []byte("12345678901234567890123456789012")
+	receiverAddress2 := []byte("12345678901234567890123456789011")
+
+	ownerNode := nodes[0]
+	initialSupply := "00" + hex.EncodeToString(big.NewInt(100000000000).Bytes())
+	scCode := arwen.GetSCCode("../../vm/arwen/testdata/erc20-c-03/wrc20_arwen.wasm")
+	scAddress, _ := ownerNode.BlockchainHook.NewAddress(ownerNode.OwnAccount.Address, ownerNode.OwnAccount.Nonce, vmFactory.ArwenVirtualMachine)
+
+	integrationTests.CreateAndSendTransactionWithGasLimit(
+		nodes[0],
+		big.NewInt(0),
+		integrationTests.MaxGasLimitPerBlock-1,
+		make([]byte, 32),
+		[]byte(arwen.CreateDeployTxData(scCode)+"@"+initialSupply))
+
+	transferTokenVMGas := uint64(6960)
+	transferTokenBaseGas := ownerNode.EconomicsData.ComputeGasLimit(&transaction.Transaction{Data: []byte("transferToken@" + hex.EncodeToString(receiverAddress1) + "@00" + hex.EncodeToString(sendValue.Bytes()))})
+	transferTokenFullGas := transferTokenBaseGas + transferTokenVMGas
+
+	initialTokenSupply := big.NewInt(1000000000)
+	initialPlusForGas := uint64(1000)
+	for _, player := range players {
+		integrationTests.CreateAndSendTransactionWithGasLimit(
+			ownerNode,
+			big.NewInt(0),
+			transferTokenFullGas+initialPlusForGas,
+			scAddress,
+			[]byte("transferToken@"+hex.EncodeToString(player.Address)+"@00"+hex.EncodeToString(initialTokenSupply.Bytes())))
+	}
+	time.Sleep(time.Second)
+
+	nrRoundsToTest := int64(5)
+	for i := int64(0); i < nrRoundsToTest; i++ {
+		round, nonce = integrationTests.ProposeAndSyncOneBlock(t, nodes, idxProposers, round, nonce)
+		integrationTests.AddSelfNotarizedHeaderByMetachain(nodes)
+
+		for _, player := range players {
+			createAndSendRelayedAndUserTx(nodes, relayer, player, scAddress, big.NewInt(0),
```

Comment:
Can you perhaps use a separate variable for the data field / param (better readability)?